src/segment-parser.js

```diff
@@ -247,7 +257,27 @@
       // used to specify some forms of timing and control data that we
       // do not currently use.
       if (afflag > 0x01) {
-        offset += data[offset] + 1;
+        adaptation_field_length = data[offset];
+
+        if (adaptation_field_length > 0) {
+          afftemp = data[offset + 1];
+
+          discontinuity_indicator = (afftemp & 0x80) >>> 7;
+          random_access_indicator = (afftemp & 0x40) >>> 6;
+          elementary_stream_priority_indicator = (afftemp & 0x20) >>> 5;
+          PCR_flag  = (afftemp & 0x10) >>> 4;
+          OPCR_flag = (afftemp & 0x08) >>> 3;
+          splicing_point_flag = (afftemp & 0x04) >>> 2;
+          transport_private_data_flag = (afftemp & 0x02) >>> 1;
+          adaptation_field_extension_flag  = afftemp & 0x01;
+
+          if (random_access_indicator === 1) {
+            if (h264Stream._h264Frame)
+              h264Stream._h264Frame.keyFrame = true;
```

Comment:
I'm not certain whether this is a good way of doing it. Another idea I have is something like `h264Stream.setNextFrameToKeyFrame()` or something.
However, this does work.